examples/evaluation_and_profiling/swe_bench/src/nat_swe_bench/predictors/predict_iterative/tools/register.py

```
@@ -0,0 +1,66 @@
+from nat.data_models.function import FunctionBaseConfig
+
+
+class GitRepoToolConfig(FunctionBaseConfig, name="git_repo_tool"):
+    """Configuration for git repository management tool."""
+    _type: typing.Literal["git_repo_tool"] = "git_repo_tool"
+    workspace_dir: str = "./.workspace"  # Base directory for cloning repositories
+    cleanup_on_exit: bool = True  # Whether to clean up repos after use
+
+
+@register_function(config_type=GitRepoToolConfig)
+async def git_repo_tool(tool_config: GitRepoToolConfig, builder: Builder):
```

Ruff (0.14.11)

Unused function argument: `builder`

(ARG001)